tools/Dashboard/DevHome.Dashboard/Views/DashboardView.xaml.cs

```diff
@@ -172,25 +152,50 @@ private async Task ConfigureWidgetRenderer(AdaptiveCardRenderer renderer)
     }
 
     private async void OnLoaded(object sender, RoutedEventArgs e)
+    {
+        await InitializeDashboard();
+    }
+
+    private bool EnsureHostingInitialized()
+    {
+        if (_widgetHostInitialized)
+        {
+            return _widgetHostInitialized;
+        }
+
+        _widgetHostInitialized = ViewModel.EnsureWebExperiencePack() && ViewModel.WidgetHostingService.GetWidgetCatalog() != null && SubscribeToWidgetCatalogEvents();
+
+        return _widgetHostInitialized;
+    }
+
+    private async Task<bool> InitializeDashboard()
     {
         LoadingWidgetsProgressRing.Visibility = Visibility.Visible;
         ViewModel.IsLoading = true;
 
-        // Cache the widget icons before we display the widgets, since we include the icons in the widgets.
-        await _widgetIconCache.CacheAllWidgetIconsAsync(_widgetCatalog);
-
-        await ConfigureWidgetRenderer(_renderer);
+        if (EnsureHostingInitialized())
+        {
+            // Cache the widget icons before we display the widgets, since we include the icons in the widgets.
+            await _widgetIconCache.CacheAllWidgetIconsAsync(ViewModel.WidgetHostingService.GetWidgetCatalog()!);

```

Comment:
NIT: If calls `CacheAllWidgetIconsAsync`, `ConfigureWidgetRenderer`, and `Restore PinnedWisgetsAsyns` don't depend on each other, consider letting these al run in parallel and using `Task.WaitAll()`